tools/query_verification/src/main/java/com/google/bigquery/BigQueryManager.java

```diff
@@ -279,8 +281,46 @@ public String getName() {
             StandardSQLTypeName type = fields.get(i).getType().getStandardType();
 
             Object result;
-            // TODO Convert value to appropriate Java type
-            result = value.getStringValue();
+            try {
+                switch (type) {
+                    case BOOL:
+                        result = value.getBooleanValue();
+                        break;
+                    case FLOAT64:
+                        result = value.getDoubleValue();
+                        break;
+                    case INT64:
+                        result = value.getLongValue();
+                        break;
+                    case NUMERIC:
+                        result = value.getNumericValue();
+                        break;
+                    case STRUCT:
+                        FieldList subFields = fields.get(i).getSubFields();
+                        FieldValueList subValues = value.getRecordValue();
+                        result = parseResults(subValues, subFields);
+                        break;
+                    case DATE:
+                        result = new SimpleDateFormat("yyyy-MM-dd").parse(value.getStringValue());
+                        break;
+                    case DATETIME:
+                        result = new SimpleDateFormat("yyyy-MM-dd'T'hh:mm:ss.SSSSSS").parse(value.getStringValue());
+                        break;
+                    case TIME:
+                        result = new SimpleDateFormat("hh:mm:ss.SSSSSS").parse(value.getStringValue());
+                        break;
+                    case TIMESTAMP:
+                        result = new SimpleDateFormat("yyyy-MM-dd hh:mm:ss.SSSSSS zzz").parse(value.getStringValue());
+                        break;
+                    case STRING:
+                        result = value.getStringValue();
+                    default:
+                        // Handle unknown/unsupported types as String
+                        result = value.getStringValue();
+                }
+            } catch (ParseException e) {
+                result = Optional.empty();
```

Comment:
Zihao did you want each object to be Optional or did you want an empty Optional added like this? I actually think null is ok in this situation.

Reply:
I think Optional.empty() looks good. My impression is java guide suggests to use optional whenever applicable.